src/typedefs.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * @typedef {import('multiaddr').Multiaddr} Multiaddr
+ * @typedef {import('multiaddr').MultiaddrObject} MultiaddrObject
+ * @typedef {import('libp2p-interfaces/src/connection').Connection} Connection
+ * @typedef {import('libp2p-interfaces/src/transport/types').Upgrader} Upgrader
+ * @typedef {import('libp2p-interfaces/src/transport/types').MultiaddrConnection} MultiaddrConnection
+ * @typedef {import('libp2p-interfaces/src/transport/types').Listener} Listener
+ * @typedef {import('net').Socket} Socket
+ * @typedef {import('net').Server & {__connections: MultiaddrConnection[]}} Server
```

Comment:
For the review comments above, I do understand know that you moved the type definitions here. However, to be consistent with the remaining libp2p codebase, we should move this to the files they are used. We decided this was the best way of understanding where the types come from, instead of having them in a different file

Reply:
And from what I can tell, this approach does not offer intellisense in visual studio code as my vscode is telling it does not know the types in the other files

Reply:
I also used VSCode, don't know why but it was able to detect the types in `typedefs` file. I will update the PR.